lib/compress/zstd_compress.c

```diff
@@ -2568,8 +2572,9 @@ static size_t ZSTD_compress_frameChunk (ZSTD_CCtx* cctx,
      * greater than the optimistic number of blocks we still have remaining.
      * This might be UNset when data is uncompressable and we're streaming.  */
 
-    const int enoughDstCapacityForNoCompressSuperBlocks =
-        (dstCapacity / (blockSize + 7 /* header + checksum */)) > (srcSize / blockSize);
+    const size_t targetCBlockSize = cctx->appliedParams.targetCBlockSize;
+    const int enoughDstCapacityForNoCompressSuperBlocks = !targetCBlockSize ? 0 :
+        (dstCapacity / (targetCBlockSize + 7 /* header + checksum */)) > (srcSize / targetCBlockSize);
```

Comment:
minor comment :
`header` is present at each block,
while `checksum` can only exist once per frame.

More accurate formula should be `(dstCapacity - 4) / (targetCBlockSize + 3)`.

This, however, requires to ensure that `dstCapacity >= 4`.

Reply:
Also :
try to avoid "magic" numbers, prefer referring to some constant defined for the project.

For example, the block size is [`ZSTD_BLOCKHEADERSIZE`](https://github.com/facebook/zstd/blob/dev/lib/common/zstd_internal.h#L127).
Sure it still means `3`, but this is more reliable for long-term maintenance (meaning is clear).

_note_ : it seems we are missing a kind of `ZSTD_FRAMECHECKSUMSIZE` constant for the second part.